homer0/projext-plugin-webpack
# refactor(project): use image-webpack-loader only when installed by the implementation

### What does this PR do?

We had a lot of issues with [`image-webpack-loader`](https://yarnpkg.com/en/package/image-webpack-loader):

- it wasn't being updated often, so it was requiring old dependencies.
- Before `3.6.0` it was like "this new version may work or not, we'll see".
- Not everyone has `libpng` installed or an friendly environment where to install it.

So the decision was made to make it _"optional"_: This plugin will no longer install it nor use it by default, but if it detects that the loader is on the implementation's `package.json` `dependencies`/`devDependencies`, it will use it the same way as before. 

### How should it be tested manually?

Install this version and create a bundle for a target that includes images, you'll see that the loader wasn't installed and no _"optimizations"_ were made to your images.

Now install the loader and bundle again; you'll see that the images are _"optimized"_ (they may loose ~1k if they're small).

And of course...

```bash
yarn test
# or
npm test
```
